### PR TITLE
support different outputs from nmap command

### PR DESCRIPTION
**Pull request by cxhong**
This is enhancement for pull request #1134 and for issue #1120 

output strings are different for nmap command on version 5.10 above than lower version.  So, the grep string has to be modified.

### CODE REVIEW

**Comment by cxhong** [comment]
@zet809 , can u review this code changes?

File: xCAT-server/lib/xcat/plugins/bmcdiscover.pm [hunk 1]
```diff
@@ -461,11 +461,11 @@ sub scan_process{
...
         my $nmap_version = $version_array[2];
-        # the output of nmap is different for version under 4.75
-        if (xCAT::Utils->version_cmp($nmap_version,"4.75") <= 0) {
-            $bcmd = join(" ",$nmap_path," -sP -n $range | grep Host |cut -d ' ' -f2 |tr -s '\n' ' ' ");
+        # the output of nmap is different for version under 5.10 
+        if (xCAT::Utils->version_cmp($nmap_version,"5.10") < 0) {
+            $bcmd = join(" ",$nmap_path," -sP -n $range | grep \"appears to be up\" |cut -d ' ' -f2 |tr -s '\n' ' ' ");
         } else {
-            $bcmd = join(" ",$nmap_path," -sn -n $range | grep for |cut -d ' ' -f5 |tr -s '\n' ' ' ");
+            $bcmd = join(" ",$nmap_path," -sn -n $range | grep \"Nmap scan report\" |cut -d ' ' -f5 |tr -s '\n' ' ' ");
```
Comment (zet809) [inline xCAT-server/lib/xcat/plugins/bmcdiscover.pm]: May check 'up' then "Nmap scan report" be better? It looks like this:
" -sn -n $range | grep -B1 up |grep \"Nmap scan report\" |cut -d ' ' -f5 |tr -s '\n' ' ' "

File: xCAT-server/lib/xcat/plugins/bmcdiscover.pm [hunk 2]
```diff
@@ -461,11 +461,11 @@ sub scan_process{
         my $version_result = xCAT::Utils->runcmd($ccmd, 0);
         my @version_array = split / /, $version_result;
         my $nmap_version = $version_array[2];
-        # the output of nmap is different for version under 4.75
-        if (xCAT::Utils->version_cmp($nmap_version,"4.75") <= 0) {
-            $bcmd = join(" ",$nmap_path," -sP -n $range | grep Host |cut -d ' ' -f2 |tr -s '\n' ' ' ");
+        # the output of nmap is different for version under 5.10
```
Comment (zet809) [inline xCAT-server/lib/xcat/plugins/bmcdiscover.pm]: The precise announcement is, from 5.10 the option '-sP' is changed to '-sn'.
It is mentioned here, https://nmap.org/changelog.html#5.10BETA1

File: xCAT-server/lib/xcat/plugins/switchdiscover.pm [hunk 3]
```diff
@@ -1259,5 +1252,30 @@ sub format_xml {
...
+      for nmap version 5.10 above, the sP option changed to sn.
+      the output of some commands also have differents.
+      example:  for snmp_scan option, 
+        version 4.75 has output "Host 10.4.25.1 appears to be up ... good."  but 
+        for version 6.40, it has output "Discovered open port 161/udp on 10.4.25.1"
+    Returns:
+      result: version of nmap on the system 
+=cut
+#--------------------------------------------------------------------------------
+
```
Comment (zet809) [inline xCAT-server/lib/xcat/plugins/switchdiscover.pm]: Why not have this function put in Utils.pm and have bmcdiscover.pm call this function to get nmap version?

**Comment by zet809** [comment]
Looks good to me, I agree to merge it.

**Pull request merged**